### ISSUE TITLE
Can't add custom fonts outside of vendor/mpdf folder

### ISSUE THREAD

**Issue by valentinocossar** [mpdf/mpdf#407]
### I have this problem/would like to have this functionality
I'm using [InvoicePlane](https://github.com/InvoicePlane/InvoicePlane) 1.5.2, that uses mPDF 7 installed via composer, so I would like to add custom fonts outside of the vendor folder. I've added custom fonts to the 'fontdata' array and changed 'fontDir' with a new path where to put all ttfonts, but neither 'fontDir' nor 'fontdata' work as expected. I inspected '$mpdf' object with var_dump after changes, but both 'fontdata' and 'fontDir' were correctly changed. Why doesn't work as expected?

P.S.
It works fine if I add the custom fonts inside the vendor/mpdf folder.
This is the file ([mpdf_helper.php](https://github.com/InvoicePlane/InvoicePlane/blob/master/application/helpers/mpdf_helper.php)) where I add the snippet provided below (starting from line 39).

### These are mPDF and PHP versions I am using
PHP: 7.0.15
mPDF: dev-development

### This is a PHP code snippet I use

```
<?php

$mpdf = new \Mpdf\Mpdf();

$mpdf->fontdata["source-sans-pro-light"] = [
  'R' => "SourceSansPro-ExtraLight.ttf",
  'B' => "SourceSansPro-Light.ttf",
];
$mpdf->fontdata["source-sans-pro-regular"] = [
  'R' => "SourceSansPro-Regular.ttf",
];
$mpdf->fontdata["source-sans-pro-bold"] = [
  'R' => "SourceSansPro-SemiBold.ttf",
  'B' => "SourceSansPro-Bold.ttf",
];

$mpdf->fontDir = FCPATH . 'uploads/temp/mpdf/ttfonts';

var_dump($mpdf);

?>
```

**Comment by lfaveur** [comment]
good question i have the same pbm i tried use it with symfony and yes it's load font from vendor folder and when i defined the folder for my fonts : 
`define('_MPDF_TTFONTPATH', $this->get('kernel')->getRootDir() . '/Resources/public/font/');`
the prboblem is that the config_fonts.php search all files who are in the array but all files are not loaded because I changed the global variable ... so what I did ? copy all font who are in 'ttfonts/' and copy in my new folder ... this is the only way i found to avoid that errors ... I don't know if it will help you @valentinocossar  or if you found a better way ...

**Comment by valentinocossar** [comment]
Hi @lfaveur, thank you for the reply. In my case, this is a little bit more complex, because InvoicePlane uses a custom version of mPDF maintained by the same maintainer of InvoicePlane. So I will provide more specific info about this issue as soon as the maintainer will update the custom version to the latest version of mPDF.

**Comment by lfaveur** [comment]
ok ... good luck for your project :)

**Comment by daudmalik06** [comment]
@valentinocossar  well there is a new library which uses chrome headless as a back-end so no more these kind of problems , give it a try very easy to install and use 
https://github.com/daudmalik06/PhpChromeToPdf

**Comment by finwe** [comment]
There was indeed an error where directories set after instantiation with fontDir were not taken into consideration.

This is fixed in fc2007f by making `fontDir` private, it is now possible to add custom directories with `AddFontDirectory` method or via `fontDir` constructor `$config` array key.
